runner/client/src/mill/runner/client/MillProcessLauncher.java

```diff
@@ -274,15 +275,54 @@ static int getTerminalDim(String s, boolean inheritError) throws Exception {
 
   private static AtomicReference<String> memoizedTerminalDims = new AtomicReference();
 
+  private static final boolean canUseNativeTerminal;
+
+  static {
+    // Load jansi native library
+    JansiLoader jansiLoader = new JansiLoader(mill.runner.client.Versions.jansiVersion());
+    File jansiLib = jansiLoader.tryLoadFast();
+    if (jansiLib == null) jansiLib = jansiLoader.loadSlow();
+
+    // We have the jansi native library, we proceed to load it.
+    System.load(jansiLib.getAbsolutePath());
+
+    // Tell jansi not to attempt to load a native library on its own
+    Class cls = org.fusesource.jansi.internal.JansiLoader.class;
+    java.lang.reflect.Field fld;
+    try {
+      fld = cls.getDeclaredField("loaded");
+    } catch (NoSuchFieldException ex) {
+      throw new RuntimeException(ex);
+    }
+    fld.setAccessible(true);
+    try {
+      fld.set(null, Boolean.TRUE);
+    } catch (IllegalAccessException ex) {
+      throw new RuntimeException(ex);
+    }
+
+    boolean canUse;
+    if (mill.constants.Util.hasConsole()) {
+      try {
+        NativeTerminal.getSize();
+        canUse = true;
+      } catch (Throwable ex) {
+        canUse = false;
+      }
+    } else canUse = false;
+
```

Comment:
Most of this logic seems like it should be moved into `JansiLoader`

Reply:
Should be fixed